ipmininet/router/config/zebra.py

```diff
@@ -155,28 +178,99 @@ def __iter__(self):
+
+    def __init__(self, type, value):
+        """
+        :param type: Type of value to me modified
+        :param value: Value to be modified
+        """
+        # TODO Check if type is correct
+        self.type = type
+        self.value = value
+
+
 class RouteMap(object):
     """A class representing a set of route maps applied to a given protocol"""
 
     # Number of route maps
     count = 0
 
-    def __init__(self, name=None, maps=(), proto=()):
-        """:param name: The name of the route-map, defaulting to rm##
-        :param maps: A set of RouteMapEntry,
+    def __init__(self, name=None, match_policy=PERMIT, match_cond=(), set_actions=(), call_action=None,
+                 exit_policy=None,
+                 order=10, proto=(), neighbor=any, direction='in'):
+        """
+        :param name: The name of the route-map, defaulting to rm##
+        :param match_policy: Deny or permit the actions if the route match the condition
+        :param match_cond: Specify one or more conditions which must be matched if the entry is to be considered further
+        :param call_action: call to an other route map
+        :param exit_policy: An entry may, optionally specify an alternative exit policy if the entry matched
                      or of (action, [acl, acl, ...]) tuples that will compose
                      the route map
-        :param proto: The set of protocols to which this route-map applies"""
+        :param order Priority of the route map compare to others
+        :param proto: The set of protocols to which this route-map applies
+        """
         RouteMap.count += 1
         self.name = name if name else 'rm%d' % RouteMap.count
-        self._entries = [e if isinstance(e, RouteMapEntry)
-                         else RouteMapEntry(action=e[0], match=e[1])
-                         for e in maps]
+        self.match_policy = match_policy
+        self.match_cond = [e if isinstance(e, RouteMapMatchCond)
+                           else RouteMapMatchCond(type=e[0], condition=e[1])
+                           for e in match_cond]
+        self.set_actions = [e if isinstance(e, RouteMapSetAction)
+                            else RouteMapSetAction(type=e[0], value=e[1])
+                            for e in set_actions]
+        self.call_action = call_action
+        self.exit_policy = exit_policy
+        self.neighbor = neighbor
+        self.direction = direction
+        self.order = order
         self.proto = proto
 
-    def __iter__(self):
-        """This Routemap is the set of all its entries"""
-        return iter(self._entries)
+    def append_match_cond(self, match_conditions):
+        """
+
+        :return:
+        """
+        for match_condition in match_conditions:
+            exist = False
+            for self_match_condition in self.match_cond:
+                exist = (match_condition.condition == self_match_condition.condition and match_condition.type == self_match_condition.type)
```

Comment:
You could implement the magic method ```__eq__()``` in RouteMapMatchCond and replace the all these lines by
```python
if match_condition not in self.match_conditions:
   self.match_cond.append(match_condition)
```